bin/libast_prereq.sh

```diff
@@ -0,0 +1,102 @@
+#!/bin/sh
+
+# This script is used for feature detection and setting up other configurations that are required to build libast
+set -x
+set -e
+
+# This script is run from an unspecified directory so we have to determine directory paths
+# http://mesonbuild.com/Reference-manual.html#run_command
+script_path=`realpath $0`
```

Comment:
Hmmm, realpath is basically a Linux thing (and would be nice to have a similar ksh93 builtin).
So IMHO ``script_path=$( cd `dirname $0`; printf "$PWD" )`` or something similar would be better, and later `_xxx_dir=/${bla_dir%/*}` instead of ``_xxx_dir=`dirname bladir` ``  (efficiency).

Reply:
@jelmd is correct. The Fish shell project had to implement a fallback builtin for `realpath` because it isn't found on all the systems supported by the project. We should avoid using it if possible.

Reply:
> Hmmm, realpath is basically a Linux thing (and would be nice to have a similar ksh93 builtin).

There is already a realpath builtin in libcmd in the beta branch.

Reply:
> There is already a realpath builtin in libcmd in the beta branch.

Yes but that isn't available when running `meson`. As painful as it is we need to do one of two things: a) mandate that */bin/bash* or */bin/ksh* be available so we can use its features, or b) assume only Bourne shell capabilities are available. The latter excludes array vars and commands like `realpath`.

Reply:
```realpath``` and ```dirname``` are available on Linux and FreeBSD. I think on macOS you can get it from coreutils package in homebrew. As I have mentioned earlier, this script is just a workaround to run iffe tests, so we should not worry too much about compatibility.

Reply:
> As I have mentioned earlier, this script is just a workaround to run iffe tests, so we should not worry too much about compatibility.

Agreed. We just need to keep this sort of thing in mind as we evolve the build process.